src/frontend/api.ts

```diff
@@ -224,29 +326,19 @@ export function Api({ baseUrl, timeout = DEFAULT_TIMEOUT }: ApiParam) {
     });
   }
 
-  // Request convenience methods that wait for the server to be ready
-  async function get(url: string) {
-    await onReady();
-    return await req.get(url).json();
-  }
-  async function del(url: string) {
-    await onReady();
-    return await req.delete(url).json();
-  }
-  async function put(url: string, data: any) {
-    await onReady();
-    return await req.put(url, { json: data }).json();
-  }
-  async function post(url: string, data: any) {
-    await onReady();
-    return await req.post(url, { json: data }).json();
-  }
-
   // Used to track RPC communication
   let channelId = 0;
 
+  const { del, get, post, put } = createRequestClient({ baseUrl, onReady });
+
   // All public methods
   const api = {
+    /**
+     * Map server methods
+     */
+    ...(devExperiments.mapSettings
+      ? { maps: createMapServerApi() }
+      : undefined),
```

Comment:
Will this work? The exported api is a single instance created at app startup. If the user turns on the setting then this will not update. Maybe keep the methods available whether the experiment is turned on or not? Otherwise would require an app restart after turning it on/off.

Reply:
`devExperiments` is the build time feature flag that enables whether or not the user even has the option to use the map server or not. the setting you're thinking of belongs to the settings context, where using the map server can be toggled. so in this case, the api becomes available if the feature flag is on, regardless of if the user has actually enabled the map server or not